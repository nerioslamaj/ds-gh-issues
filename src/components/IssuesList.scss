@import '../index.scss';

.Issues-list-wrapper {
    margin: 32px 0;

    .Issues-list {
        border-radius: 8px;
        overflow: hidden;
    }

    .Issues-pagination {
        color: $light-txt-2;
        font-size: 1.2rem;
        font-weight: 700;
        text-align: center;
        margin-top: 32px;

        &.disabled {
            display: none;
        }

        span {
            cursor: pointer;
            &.disabled {
                color: $disabled-txt;
                cursor: default;
            }
        }
    }
}