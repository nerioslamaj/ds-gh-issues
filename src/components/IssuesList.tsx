import React from 'react';
import axios from 'axios'
import IssueItem from './IssueItem'
import './IssuesList.scss';
import GLOBAL from '../global';

type IssuesListState = {
  issuesList: Array<any>,
  currentPage: number,
  maxPage: number
};

class IssuesList extends React.Component<any, IssuesListState> {
  constructor(props: any) {
    super(props);
    this.state = { 
      issuesList: [],
      currentPage: 1,
      maxPage: 4
    };
  }

  componentDidMount() {
    this.getIssuesList(this.state.currentPage);
  }

  async getIssuesList(page: number) {
    if(page > 0 && page <= this.state.maxPage) {
      const fullUrl = `${GLOBAL.issuesAPI}?page=${page}`;
  
      const res = await axios.get(fullUrl);
      const data: Array<any> = await res.data;
  
      this.setState({ 
        issuesList: data,
        currentPage: page
      })
    }
  }

  render() {
    return (
      <div className="Issues-list-wrapper">
        <div className="Issues-list">
          {this.state.issuesList.map(function (issue) {
            return <IssueItem
              key={issue.number}
              title={issue.title}
              author={issue.user.login}
              number={issue.number}
              createdAt={issue.created_at}
              labels={issue.labels}
              url={issue.html_url}
            />;
          })}
        </div>
        <p className="Issues-pagination">
          <span
            className={this.state.currentPage === 1 ? 'disabled' : ''}
            onClick={async () => {await this.getIssuesList(this.state.currentPage - 1);} }
          >Previous </span>
          â€”
          <span
            className={this.state.currentPage === this.state.maxPage ? 'disabled' : ''}
            onClick={async () => {await this.getIssuesList(this.state.currentPage + 1);} }
          > Next</span>
        </p>
      </div>
    );
  }
}

export default IssuesList;
