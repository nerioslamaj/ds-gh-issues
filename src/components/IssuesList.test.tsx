import React from 'react';
import IssuesList from './IssuesList';
import TestRenderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = TestRenderer
    .create(<IssuesList></IssuesList>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
