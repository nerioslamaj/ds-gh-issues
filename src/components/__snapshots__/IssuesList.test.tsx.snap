// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Issues-list-wrapper"
>
  <div
    className="Issues-list"
  />
  <p
    className="Issues-pagination disabled"
  >
    <span
      className="disabled"
      onClick={[Function]}
    >
      Previous 
    </span>
    â€”
    <span
      className=""
      onClick={[Function]}
    >
       Next
    </span>
  </p>
</div>
`;
